reactor-core/src/main/java/reactor/core/publisher/FluxGroupBy.java

```diff
@@ -224,9 +224,9 @@ public void onComplete() {
 			}
 			for (UnicastGroupedFlux<K, V> g : groupMap.values()) {
 				g.onComplete();
+				GROUP_COUNT.decrementAndGet(this);
```

Comment:
Actually, this decrement is not really necessary as the upstream won't procude further items and thus cancellation has no effect.